CodeReviewStandards/rfc-008-review-checklist.md

```diff
@@ -0,0 +1,44 @@
+# Pull Request Standards
+
+## Summary
+
+A checklist designed to help facilitate code reviews by outlining a pre-review checklist to follow for the raiser of the pull request, and a reviewer checklist.
+
+## Problem
+
+Pull requests follow no set format and vary between projects, which in turn causes reviews to take longer and require more context switching. Code review and commentary often happens 'offline', which results in reduced context and fewer points of reference when reviewing changes later.
+
+## Proposal
+
+Before raising a pull request, the pull request SHOULD meet the following criteria:
+ - [ ] Pull request description outlines _what has changed_
+ - [ ] Pull request description outlines _why the change was made_
+ - [ ] If the project has a linter, it SHOULD not report any issues with the pull request
+ - [ ] Code follows language specific conventions
+ - [ ] Code follows project specific conventions
+ - [ ] Code has been refactored with quality and readability in mind
+ - [ ] Tests are obviously the product of practising TDD (where appropriate)
+ - [ ] There are high level feature/acceptance tests in place as appropriate
+ - [ ] Tests have been refactored - aim for quality, readability and maintainability
+ - [ ] Tests all pass and the change satisfies the goal of _why the change was made_
```

Comment:
I sometimes (like all the time) leave high-level tests pending in order to do small PRs, I like the fact it also shows what the code will eventually have to deal with

Reply:
What do you suggest? To go to production, you'd need a sane test suite that shows the code satisfies all requirements, but I can appreciate that's not a requirement for all PRs. Remove that check? Reword to something like 'Test suite shows that code is tested to adequate confidence' etc?

Reply:
Seems reasonable to me

Reply:
How about marking those tests as wip so they don't fail when running the suite? This can be achieved in most test frameworks.